### ISSUE TITLE
Some issues

### ISSUE THREAD

**Issue by matthijskooijman** [LacunaSpace/basicmac#5]
Hello,

The last time I compiled your code in the Arduino library.
https://github.com/ricaun/arduino-BasicMAC

The sx127x works ok but the sx126x I have some problem with the Dio3. My board does not have the DIO3 to drive connected to the crystal, and the code forces to use dio3 on `SetDIO3AsTcxoCtrl `.

I was trying to use US915 but the code has some bug on a `REG_FIX`, the join does not work very well.

I noticed the original code has updated to version a final release 2.2
https://github.com/lorabasics/basicmac

And a lot of the code change, like the `nextJoinState` has some update and the `SetDIO3AsTcxoCtrl `does not exist anymore.

See yaa.

_Originally posted by @ricaun in https://github.com/LacunaSpace/basicmac/pull/2#issuecomment-614338777_

**Comment by matthijskooijman** [comment]
> The sx127x works ok but the sx126x I have some problem with the Dio3. My board does not have the DIO3 to drive connected to the crystal, and the code forces to use dio3 on SetDIO3AsTcxoCtrl .

Yeah, that should probably be configurable. But is it problematic for you?  Do you have anything else connected to DIO3? Otherwise it would just toggle the pin and wait for no reason, but it should still work?

> I was trying to use US915 but the code has some bug on a REG_FIX, the join does not work very well.

Hm, I indeed did not test US915 at all, I'll add a note to the README. If you figure out what is wrong, contributions are welcome :-)

> I noticed the original code has updated to version a final release 2.2
lorabasics/basicmac**

Bugger, someone else also just pointed that out. Should have included that in rebaseing push yesterday. Now I'll try to merge this on top rather than rebasing it below (which might be better anyway). There's a lot of small changes in a huge commit, including whitespace-only changes, so this might get me a ton of conflicts, but we'll see :-) I created #6 to track this.

**Comment by matthijskooijman** [comment]
For the DIO3 issue, I've created #9.

For US915 and other regions, I've created #8.

For the 2.2 update, I've created and since fixed #6, so I'm going to close this issue.

> And a lot of the code change, like the nextJoinState has some update and the SetDIO3AsTcxoCtrl does not exist anymore.

For completeness: `SetDIO3AsTcxoCtrl` never existed in the Semtech 2.1 version, I added that version in this repo (so it was not removed in 2.2, it was just never added there).
